File: rest/src/connection/pg_connection.ts
import { config } from "../utils/config.ts";
import { postgres } from "../../deps.ts";

// https://deno.land/x/postgres@v0.12.0

const dbParams = {
    user: config.database.pg_user,
    database: config.database.pg_db,
    hostname: config.database.pg_host,
    port: config.database.pg_port,
    password: config.database.pg_pass,
};

const pool = new postgres.Pool(dbParams, config.database.max_pool_connections, true);

async function runQuery(query: string, _pool: postgres.Pool = pool) {
    const client = await _pool.connect();
    try {
        const result = await client.queryObject(query);
        return result;
    } finally {
        client.release();
    }
}

export {
    runQuery
}

File: migration/src/connection/pg_connection.py
import psycopg2

# utils.environment swaps environment variables on import
from utils.config import DATABASE_CONFIG

POSTGRES_USER = DATABASE_CONFIG["pg_user"]
POSTGRES_PASS = DATABASE_CONFIG["pg_pass"]
POSTGRES_DATABASE = DATABASE_CONFIG["pg_db"]
POSTGRES_HOST = DATABASE_CONFIG["pg_host"]
POSTGRES_PORT = DATABASE_CONFIG["pg_port"]

def make_pg_pool():
    return psycopg2.connect(
        database=POSTGRES_DATABASE,
        user=POSTGRES_USER,
        password=POSTGRES_PASS,
        host=POSTGRES_HOST,
        port=POSTGRES_PORT,
    )